### PR TITLE
FEAT Use parquet files as output

### PR DESCRIPTION
**Pull request by GeorgWa**
This PR changes the default behavior of alphaDIA to use binary parquet files for all intermediate outputs.
Furthermore, the user get's an option in `config['search_output']['file_format']` to receive all parquet outputs

### CODE REVIEW

File: alphadia/outputaccumulator.py [hunk 1]
```diff
@@ -121,8 +122,8 @@ def parse_output_folder(
 
 
         """
-        psm_df = pd.read_csv(os.path.join(folder, "psm.tsv"), sep="\t")
-        frag_df = pd.read_csv(os.path.join(folder, "frag.tsv"), sep="\t")
+        psm_df = pd.read_parquet(os.path.join(folder, "psm.parquet"))
```
Comment (jalew188) [inline alphadia/outputaccumulator.py:125]: Comparing to tsv, parquet is better. Have you considered .hdf as the intermediate file? What is the cons and progs of parquet comparing to .hdf?

Comment (jalew188) [reply alphadia/outputaccumulator.py:125]: pd.read_parquet may be the main reason?

Comment (GeorgWa) [reply alphadia/outputaccumulator.py:125]: Yes, the main reasons are 1) immediate support by pandas and 2) adoption by the MS community as .parquet is now used in Sage and DIANN 👍🏻

File: alphadia/consensus/utils.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    logger.info(f"Reading {file_path} from disk")
+
+    if file_format == "parquet":
+        return pd.read_parquet(file_path)
+
+    elif file_format == "tsv":
+        return pd.read_csv(file_path, sep="\t")
+
+    else:
+        raise ValueError("I don't know how you ended up here")
```
Comment (mschwoer) [inline alphadia/consensus/utils.py:48]: me neither, maybe bc it's redundant to l.29 ;-)
can be removed I guess .. or you move the logic from l.29 down here..

File: alphadia/consensus/utils.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    path_no_format: str
+        Path for file without format
+
+    file_format: str, default = 'parquet'
+        File format for loading the file. Available options: ['parquet', 'tsv']
+
+    """
+
+    if file_format not in supported_formats:
```
Comment (mschwoer) [inline alphadia/consensus/utils.py:68]: same as above: move down

File: alphadia/constants/default.yaml [hunk 4]
```diff
@@ -129,6 +129,8 @@ search_output:
   num_samples_quadratic: 50
   min_nonnan: 3
   normalize_lfq: True
+  # can be either "parquet" or "tsv"
+  file_format: "parquet"
```
Comment (mschwoer) [inline alphadia/constants/default.yaml:133]: shall we really make parquet the default already? maybe wait until it's tested a bit more?

File: alphadia/outputtransform.py [hunk 5]
```diff
@@ -54,22 +54,14 @@ def get_frag_df_generator(folder_list: List[str]):
 
     for folder in folder_list:
         raw_name = os.path.basename(folder)
-        frag_path = os.path.join(folder, "frag.tsv")
+        frag_path = os.path.join(folder, "frag.parquet")
```
Comment (mschwoer) [inline alphadia/outputtransform.py:57]: these file names could be gathered in some central constants class

File: alphadia/outputtransform.py [hunk 6]
```diff
@@ -596,11 +578,10 @@ def build_precursor_table(
...
         if save:
             logger.info("Writing precursor output to disk")
-            psm_df.to_csv(
-                os.path.join(self.output_folder, f"{self.PRECURSOR_OUTPUT}.tsv"),
-                sep="\t",
-                index=False,
-                float_format="%.6f",
+            write_df(
+                psm_df,
+                os.path.join(self.output_folder, f"{self.PRECURSOR_OUTPUT}"),
```
Comment (mschwoer) [inline alphadia/outputtransform.py:583]: f"" not need I guess (also below)

File: alphadia/workflow/peptidecentric.py [hunk 7]
```diff
@@ -1152,13 +1156,13 @@ def _build_candidate_speclib_flat(
         Dataframe with scored candidates
     """
     # remove decoys
-    psm_df = psm_df[psm_df["decoy"] == 0]
+    # psm_df = psm_df[psm_df["decoy"] == 0]
 
-    for col in ["rt_observed", "mobility_observed", "mz_observed"]:
-        optional_columns += [col] if col in psm_df.columns else []
+    # make copy to avoid modifying the original dataframe
+    _optional_columns = [col for col in optional_columns if col in psm_df.columns]
```
Comment (mschwoer) [inline alphadia/workflow/peptidecentric.py:1162]: why the underscore?

File: tests/unit_tests/conftest.py [hunk 8]
```diff
@@ -13,6 +14,7 @@
 
 def mock_precursor_df(
     n_precursor: int = 100,
+    with_decoy=True,
```
Comment (mschwoer) [inline tests/unit_tests/conftest.py:17]: please adapt docstring

File: tests/unit_tests/conftest.py [hunk 9]
```diff
@@ -181,3 +185,14 @@ def pytest_configure(config):
         pytest.test_data[raw_folder] = raw_files
 
     # important to supress matplotlib output
+
+
+def random_tempfolder():
```
Comment (mschwoer) [inline tests/unit_tests/conftest.py:190]: docstring?

File: tests/unit_tests/conftest.py [hunk 10]
```diff
@@ -181,3 +185,14 @@ def pytest_configure(config):
...
+    tempdir = tempfile.gettempdir()
+    # 6 alphanumeric characters
+    random_foldername = "alphadia_" + "".join(
```
Comment (mschwoer) [inline tests/unit_tests/conftest.py:193]: `alphadia_temp_` ?
also, maybe log the name

**Review by mschwoer — APPROVED** [review]
LGTM

File: alphadia/constants/default.yaml (hunk 4 above)
Comment (GeorgWa) [reply alphadia/constants/default.yaml:133]: good point

File: alphadia/workflow/peptidecentric.py (hunk 7 above)
Comment (GeorgWa) [reply alphadia/workflow/peptidecentric.py:1162]: already changed in other PR

**Pull request merged**